src/platform_impl/windows/event_loop.rs

```diff
@@ -265,10 +265,37 @@ impl<T> EventLoopWindowTarget<T> {
     }
 }
 
+/// Returns the id of the main thread.
+///
+/// Windows has no real API to check if the current executing thread is the "main thread" unlike macOS.
+/// Unix-like OSes have some APIs or semantics that exist regarding whether the main thread is in use or
+/// if the main thread id equals the pid like on Linux.
+///
+/// Windows will let us look up the current thread's id but this leaves us with the question of what
+/// is the main thread's id?
+/// We would somehow need to get the main thread's id before a developer could spin off any other
+/// threads inside of the main entrypoint in order to emulate the capabilities of other platforms.
+///
+/// We can get the id of the main thread by using CRT initialization.
+/// CRT initialization can be used to setup global state within a program by calling a list of
+/// function pointers which should assign a value to a static field.
```

Comment:
This makes it sound like we're the ones calling a list of function pointers.

Reply:
OS calls the function pointers. New docs there should make it more clear.